HUDOC.js

```diff
@@ -0,0 +1,502 @@
+		&& !(text(doc, "title").toLowerCase().includes("translation]") // toLowerCase() is added because "translation" is sometimes capitalized
+			|| docType.includes("Press Release"))) {
+			return "case";
+		}
+	}
+
+
+	return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "case") {
+		scrapeDecision(doc, url);
+	}
+}
+
+function scrapeDecision(doc, url) { // Works for both Court judgments and decisions
+	var item = new Zotero.Item("case");
+
+	// Title
+	var capTitle = ZU.capitalizeTitle(text(doc, "title").toLowerCase(), true);
+
+	var splitTitle = capTitle.split(/\(/);
+
+	// If title already contains the type, remove it.
+	// Exclude non-split titles and advisory opinions
+	if (!(splitTitle.length == 1) &&
+			!scrapeMetaData(doc, "typedescription").includes("Advisory Opinion")) {
+
+		let titleVersusMatch = splitTitle.findIndex(function(elem) {
+			if (elem.includes("V.") || elem.includes("C.")) {
+						return true;
+			}
+			return false;
+		});
+
+		let titleNumMatch = splitTitle.findIndex(function(elem) {
+			if (/^(no.|II|IV|V\)|VI|IX|X\))/i.test(elem)) {
+						return true;
+			}
+			return false;
+		});
+
+		let titleIndex = Math.max(titleVersusMatch, titleNumMatch);
+
+
+		capTitle = splitTitle.slice(0, titleIndex+1).join("(");
+	}
+
+	// Zotero capitalizes the "v."/"c.", so we need to correct that for English and French cases
+	if (capTitle.includes(" V. ")) {
+		capTitle = capTitle.replace(" V. ", " v. ");
+	}
+	else if (capTitle.includes(" C. ")) {
+		capTitle = capTitle.replace(" C. ", " c. ");
+	}
+	item.caseName = capTitle + getTypeBit(doc, url);
+
+	// Court
+	var court = scrapeMetaData(doc, "originatingbody");
+
+	if (url.includes("hudoc.echr.coe.int/fre#")) {
+		item.court = "ECtHR";
+	}
+	else {
+		if (text(doc, "title").includes("Advisory Opinion")) {
+			item.court = "ECtHR [GC]";
+		}
+		else if (court.includes("Grand Chamber")) {
+			item.court = "ECtHR [GC]";
+		}
+		else if (court.includes("Commission")) {
+			item.court = "Commission";
+		}
+		else if (court == "Committee of Ministers") {
+			item.court = "Committee of Ministers";
+		}
+		else {
+			item.court = "ECtHR";
+		}
+	}
+
+	// Date of decision
+	// convert to simple ISO: yyyy-mm-dd dd.mm.yyyy.
+	item.dateDecided = scrapeMetaData(doc, "judgementdate").split("/").reverse().join('-');
+
+	item.url = "http://hudoc.echr.coe.int/eng?i=" + getItemID(url);
+
+	// Query remaining metadata from API
+	var queryUrl = "https://hudoc.echr.coe.int/app/query/results?query=(contentsitename=ECHR) AND "
+					+ getItemID(url)
+					+ "&select=appno,conclusion,docname,languageisocode" // Adapt based on what is needed
+					+ "&sort=&start=0&length=1";
+
+	ZU.doGet(queryUrl, function (json) {
+		json = JSON.parse(json).results[0].columns;
+
+		Zotero.debug("Queried HUDOC API at: " + queryUrl);
+
+		// Application numbers
+		// FIXME: Decide whether to prepend "app. no.". Some styles automatically add "no." (Chicago)
```

Comment:
This should be the style's concern, not the translator's. Appending text to numeric fields prevents localization.

Reply:
Removed app no logic